Tools/TestWebKitAPI/Tests/WTF/WorkQueue.cpp

```diff
@@ -361,17 +363,13 @@ TEST(WTF_WorkQueue, ThreadSafetyAnalysisAssertIsCurrentWorks)
 #endif
 TEST(WTF_WorkQueue, MAYBE_MainWorkQueueIsCurrent)
 {
-    assertIsCurrent(WorkQueue::main());
+    EXPECT_TRUE(WorkQueue::main().isCurrent());
```

Comment:
Here too, the better and intended assertion is the `assertIsCurrent(WorkQueue::main());`.
So you get the thread safety analysis should the future variable declarations start using it.